src/OrbitGl/ThreadStateBar.cpp

```diff
@@ -206,9 +208,33 @@ std::string ThreadStateBar::GetThreadStateSliceTooltip(PrimitiveAssembler& primi
   uint64_t end_ns = thread_state_slice->end_timestamp_ns();
   tooltip += absl::StrFormat(
       "<br/>"
-      "<b>Time:</b> %s",
+      "<b>Time:</b> %s"
+      "<br/>",
       orbit_display_formats::GetDisplayTime(TicksToDuration(begin_ns, end_ns)));
 
+  if (thread_state_slice->switch_out_or_wakeup_callstack_id().has_value()) {
+    const orbit_client_data::CallstackData& callstack_data = capture_data_->GetCallstackData();
+    const uint64_t callstack_id = thread_state_slice->switch_out_or_wakeup_callstack_id().value();
+    const orbit_client_data::CallstackInfo* callstack = callstack_data.GetCallstack(callstack_id);
+
+    static const std::string unknown_return_text = "Function call information missing";
+
+    tooltip += "<b>Triggering callstack:</b><br/>";
```

Comment:
We need to improve how we display whether the callstack comes from the thread itself or another thread.
For example (just an idea, we probably want to refine this, I need to see it in the tooltip):
"Unblocking callstack from thread <thread_name> [tid] of process <process_name> [pid]:"
"Blocking callstack from this thread:"

Reply:
Also we should make sure to convey that the callstacks correspond to the beginning of the slice. For example (again, just throwing ideas):
"This thread switched to the <name of the state> state when thread **<thread_name>** [tid] of process **<process_name>** [pid] hit/executed the following callstack:"/
"This thread switched to the <name of the state> state on hitting/executing the following callstack:"

Reply:
Edit: markdown formatting had hidden some stuff.

Reply:
I kind of like the last one most. In case the callstack comes from another thread, I think we already display this pretty clearly. I goess the most important part is that the callstack origins from the beginning of the slice.

Reply:
See new screenshot: https://screenshot/Ag6j974BZXHjPaB

Reply:
The last two were not alternatives. They were for the two cases where the callstack is from another thread or the same thread. So in the screenshot the other version should have appeared: "This thread switched to the <name_of_the_state> state when thread <thread_name> [tid] of process <process_name> [pid] hit/executed the following callstack:"

> In case the callstack comes from another thread, I think we already display this pretty clearly.

I strongly disagree. There is no indication that "callstack" and "unblocked by" are connected.

Reply:
Saying "This thread switched to the <name_of_the_state> state on hitting/executing the following callstack:" is supposed to suggest that the callstack is on the same thread as the "subject" of "executing" would be "this thread".

Reply:
Agree and done.

Reply:
Could I get a final screenshot?

Reply:
Runnable:
http://screenshot/4Htrnjt6okAWts9

Sleep:
http://screenshot/6Bdjg6grgQubqEZ